staging/cse/windows/azurecnifunc.ps1

```diff
@@ -93,6 +93,30 @@ function Set-AzureCNIConfig
         $configJson.plugins[0].AdditionalArgs[1].Value.DestinationPrefix = $KubeServiceCIDR
     }
 
+    # Replace OutBoundNAT with LoopbackDSR for IMDS acess if AKS cluster disabled Windows OutBoundNAT.
+    # The Azure Instance Metadata Service (IMDS) provides information about currently running virtual machine instances.
+    # IMDS is a REST API that's available at a well-known, non-routable IP address (169.254.169.254)
+    # Details: https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service?tabs=windows#known-issues-and-faq
+    if ($global:IsDisableWindowsOutboundNat)
+    {
+        $valueObj = [PSCustomObject]@{
+            Type = 'LoopbackDSR'
+            IPAddress = '169.254.169.254'
+        }
+        $jsonContent = [PSCustomObject]@{
+            Name = 'EndpointPolicy'
+            Value = $valueObj
+        }
+
+        # AdditionalArgs[0] is OutBoundNAT with ExceptionList; AdditionalArgs[1] is Route.
+        # Disable OutBoundNAT&Route if WinDSR is enabled. Otherwise, only disable OutBoundNAT and reserve Route.
+        if ($global:KubeproxyFeatureGates.Contains("WinDSR=true")) {
+            $configJson.plugins[0].AdditionalArgs = @($jsonContent)
+        } else {
+            $configJson.plugins[0].AdditionalArgs[0] = $jsonContent

```

Comment:
It will remove other settings, for example, settings when IsDualStackEnabled is true

Reply:
From the line 63-74, AdditionalArgs is always an array.
```
    $osBuildNumber = (get-wmiobject win32_operatingsystem).BuildNumber
    if ($osBuildNumber -le 17763){
        # In WS2019 and below rules in the exception list are generated by dropping the prefix lenght and removing duplicate rules.
        # If multiple execptions are specified with different ranges we should only include the broadest range for each address.
        # This issue has been addressed in 19h1+ builds

        $processedExceptions = GetBroadestRangesForEachAddress $exceptionAddresses
        Write-Log "Filtering CNI config exception list values to work around WS2019 issue processing rules. Original exception list: $exceptionAddresses, processed exception list: $processedExceptions"
        $configJson.plugins.AdditionalArgs[0].Value.ExceptionList = $processedExceptions
    }
    else {
        $configJson.plugins.AdditionalArgs[0].Value.ExceptionList = $exceptionAddresses
    }
```

Reply:
Split this logic from this PR.